pkg/host-disk/host-disk_test.go

```diff
@@ -148,19 +126,25 @@ var _ = Describe("HostDisk", func() {
 		Context("With multiple HostDisk volumes", func() {
 			Context("With non existing disk.img", func() {
 				It("Should create disk.img if there is enough space", func() {
-					By("Creating a new minimal vmi")
-					vmi := api.NewMinimalVMI("fake-vmi")
-
-					By("Adding a HostDisk volumes")
-					addHostDisk(vmi, "volume1", v1.HostDiskExistsOrCreate, "64Mi")
-					addHostDisk(vmi, "volume2", v1.HostDiskExistsOrCreate, "128Mi")
-					addHostDisk(vmi, "volume3", v1.HostDiskExistsOrCreate, "80Mi")
+					By("Creating a new VMI with HostDisk volumes")
+					vmi := libvmi.New(
+						libvmi.WithHostDisk("volume1", createHostDisk("volume1"), v1.HostDiskExistsOrCreate),
+						libvmi.WithHostDisk("volume2", createHostDisk("volume2"), v1.HostDiskExistsOrCreate),
+						libvmi.WithHostDisk("volume3", createHostDisk("volume3"), v1.HostDiskExistsOrCreate),
+					)
+					By("Setting the capacity for each HostDisk volume")
+					vmi.Spec.Volumes[0].HostDisk.Capacity = resource.MustParse("64Mi")
+					vmi.Spec.Volumes[1].HostDisk.Capacity = resource.MustParse("128Mi")
+					vmi.Spec.Volumes[2].HostDisk.Capacity = resource.MustParse("80Mi")
 
 					By("Executing CreateHostDisks which should create disk.img")
 					err := hostDiskCreator.Create(vmi)
 					Expect(err).NotTo(HaveOccurred())
 
-					// check if images exist and the size is adequate to requirements
+					// Ensure volumes are not nil and correctly initialized
+					Expect(vmi.Spec.Volumes).NotTo(BeNil())
```

Comment:
Please drop this. If `vmi.Spec.Volumes` were nil, you got a nil pointer dereference error during:
```
vmi.Spec.Volumes[0].HostDisk.Capacity ...
```